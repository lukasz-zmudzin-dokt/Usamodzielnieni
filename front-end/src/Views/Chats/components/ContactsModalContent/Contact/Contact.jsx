import React from 'react';
import { Row, Col, Button } from 'react-bootstrap';
import { IndexLinkContainer } from 'react-router-bootstrap';


const Contact = ({ contact }) => {
    return (
        <Row>
            <Col>
                <h5>{contact.firts_name}</h5>
            </Col>
            <Col>
                <IndexLinkContainer to={`/chats/${contact.id}`}>
                    <Button>Nowa wiadomość</Button>
                </IndexLinkContainer>
            </Col>
        </Row>
    )
}

export default Contact