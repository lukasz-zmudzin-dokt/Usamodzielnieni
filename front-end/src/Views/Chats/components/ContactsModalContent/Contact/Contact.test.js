import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent, waitForElement } from "@testing-library/react";
import Contact from "./Contact";

describe("Contact", () => {
  const contact = {
    first_name: "imie",
    last_name: "nazwisko",
    id: "1234",
    role: "goÅ›ciu",
  };
  it("should render without crashing", () => {
    const { getByText } = render(
      <MemoryRouter>
        <Contact contact={contact} />
      </MemoryRouter>
    );
    expect(getByText("imie", { exact: false })).toBeInTheDocument();
  });

  it("should match snapshot", async () => {
    const { container } = render(
      <MemoryRouter>
        <Contact contact={contact} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
