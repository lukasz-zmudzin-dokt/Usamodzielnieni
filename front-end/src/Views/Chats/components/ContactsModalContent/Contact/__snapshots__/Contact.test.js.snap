// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match snapshot 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h5>
          imie
        </h5>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h5>
          nazwisko
        </h5>
      </div>
      <div
        class="col"
      >
        <a
          class="float-right btn btn-primary"
          href="/chats/1234"
        >
          -&gt;
        </a>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <em>
          go≈õciu
        </em>
      </div>
    </div>
  </div>
</div>
`;
